.github/workflows/main.tf

```diff
@@ -0,0 +1,35 @@
+provider "aws" {
+  region = var.aws_region
+}
+
+terraform {
+  backend "s3" {
+    bucket         = "your-s3-bucket-name"
+    key            = "terraform.tfstate"
+    region         = var.aws_region
+    encrypt        = true
+    dynamodb_table = "terraform-lock"
+  }
+}
+
+resource "aws_instance" "ec2_instance" {
+  ami           = var.ami_id
+  instance_type = var.instance_type
+  key_name      = var.key_name
+
+  tags = {
+    Name = "MyEC2Instance"
+  }
+}
+
+  user_data = <<-EOF
+    #!/bin/bash
+    ssh -i ~/my_pem.pem ubuntu@current_instance_ip
```

Comment:
@shihgianlee we'd need to fill in the `current_instance_ip` should we store this in a secret and pass template it the value?

Reply:
we can store it in a github secret and pass in the value

Reply:
https://docs.github.com/en/actions/security-guides/encrypted-secrets

Reply:
A quick question on Github Action. Can we limit to LG contributors to execute the Github Action? We don't want any non LG users to have the ability to execute the pipeline. Is there a permission setting on Github for that?